cmd/tailscale/cli/serve.go

```diff
@@ -709,3 +709,19 @@ func (e *serveEnv) runServeFunnel(ctx context.Context, args []string) error {
 	}
 	return nil
 }
+
+// checkHasAccess checks three things: 1) an invite was used to join the
+// Funnel alpha; 2) HTTPS is enabled; 3) the node has the "funnel" attribute.
+// If any of these are false, an error is returned describing the problem.
+func checkHasAccess(capabilities []string) error {
+	if !slices.Contains(capabilities, "https://tailscale.com/cap/warn-funnel-no-invite") {
+		return errors.New("Funnel not available; an invite is required to join the alpha. See https://tailscale.com/kb/1223/tailscale-funnel/.")
+	}
+	if !slices.Contains(capabilities, "https://tailscale.com/cap/warn-funnel-no-https") {
```

Comment:
not not. Just "if slices.Contains..." on both these /warn ones.

Want to move them into tailcfg constants while you're here so then control can depend on them later?

Reply:
Too much 🦃 🍗  Thanks! Will make the constants.

Reply:
PTAL